pkg/api/types.go

```diff
@@ -799,6 +799,22 @@ type NodeCondition struct {
 	Message            string            `json:"message,omitempty"`
 }
 
+type NodeAddressKind string
+
+// These are valid address type of node. NodeLegacyHostIP is used to transit from out-dated
+// HostIP field to NodeAddress.
+const (
+	NodeLegacyHostIP NodeAddressKind = "LegacyHostIP"
+	NodeHostName     NodeAddressKind = "Hostname"
+	NodeExternalIPv4 NodeAddressKind = "ExternalIPv4"
```

Comment:
Hmm, just want to be able to do a quicker lookup.  But agreed ip version is too verbose here. I'll remove it.